b2sdk v2 delete_bucket() throws v3.BucketIdNotFound exception

This is especially problematic, as _v3 is still unreleased, it's hard to catch in a manner that won't break on next b2sdk update.
That case is supposed to be handled by current code, but somehow error translation still fails (tested against current master branch).

```
../b2-sdk-python/b2sdk/api.py:386: in delete_bucket
    self.session.delete_bucket(account_id, bucket.id_)
../b2-sdk-python/b2sdk/session.py:183: in delete_bucket
    return self._wrap_default_token(self.raw_api.delete_bucket, account_id, bucket_id)
../b2-sdk-python/b2sdk/session.py:456: in _wrap_default_token
    return self._wrap_token(raw_api_method, TokenType.API, *args, **kwargs)
../b2-sdk-python/b2sdk/session.py:462: in _wrap_token
    return self._reauthorization_loop(partial_callback)
../b2-sdk-python/b2sdk/session.py:477: in _reauthorization_loop
    return callback()
../b2-sdk-python/b2sdk/session.py:467: in _api_token_callback
    return raw_api_method(api_url, account_auth_token, *args, **kwargs)
../b2-sdk-python/b2sdk/raw_api.py:530: in delete_bucket
    bucketId=bucket_id
../b2-sdk-python/b2sdk/raw_api.py:458: in _post_json
    return self.b2_http.post_json_return_json(url, headers, params)
../b2-sdk-python/b2sdk/b2http.py:309: in post_json_return_json
    _timeout=timeout,
../b2-sdk-python/b2sdk/b2http.py:262: in post_content_return_json
    response = self._translate_and_retry(do_post, try_count, post_params)
../b2-sdk-python/b2sdk/b2http.py:507: in _translate_and_retry
    return cls._translate_errors(fcn, post_params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'b2sdk.v2.b2http.B2Http'>
fcn = <function B2Http.post_content_return_json.<locals>.do_post at 0x7f9cd89ca5f0>
post_params = {'accountId': 'censored', 'bucketId': '2a71a86d67b1d13b83b50e14'}

    @classmethod
    def _translate_errors(cls, fcn, post_params=None):
        """
        Call the given function, turning any exception raised into the right
        kind of B2Error.
    
        :param dict post_params: request parameters
        """
        response = None
        try:
            response = fcn()
            if response.status_code not in [200, 206]:
                # Decode the error object returned by the service
                error = json.loads(response.content.decode('utf-8')) if response.content else {}
                extra_error_keys = error.keys() - ('code', 'status', 'message')
                if extra_error_keys:
                    logger.debug(
                        'received error has extra (unsupported) keys: %s', extra_error_keys
                    )
                raise interpret_b2_error(
                    int(error.get('status', response.status_code)),
                    error.get('code'),
                    error.get('message'),
                    response.headers,
>                   post_params,
                )
E               b2sdk.exception.BucketIdNotFound: Bucket with id=2a71a86d67b1d13b83b50e14 not found

../b2-sdk-python/b2sdk/b2http.py:436: BucketIdNotFound

```

This doesn't quite seem "apiver" legal (more like accidental export to public api), but to catch these I'm using `from b2sdk.v2.exception import v3BucketIdNotFound` now.

--- a/b2sdk/v2/b2http.py
+++ b/b2sdk/v2/b2http.py
@@ -17,7 +17,7 @@ from .exception import BucketIdNotFound
 # Overridden to retain old-style BadRequest exception in case of a bad bucket id
 class B2Http(v3.B2Http):
     @classmethod
-    def _translate_error(cls, fcn, post_params=None):
+    def _translate_errors(cls, fcn, post_params=None):
         try:
             return super()._translate_errors(fcn, post_params)
         except v3BucketIdNotFound as e:

--- a/changelog.d/437.fixed.md
+++ b/changelog.d/437.fixed.md
@@ -1,0 +1,1 @@
+Ensure `b2sdk.v2.b2http` emits `b2sdk.v2.BucketIdNotFound` exception instead of `b2sdk._v3.BucketIdNotFound`.

--- a/test/unit/b2http/test_b2http.py
+++ b/test/unit/b2http/test_b2http.py
@@ -38,24 +38,24 @@ from b2sdk._internal.b2http import setlocale
 from ..test_base import TestBase
 
 
-class TestTranslateErrors(TestBase):
+class TestTranslateErrors:
     def test_ok(self):
         response = MagicMock()
         response.status_code = 200
         actual = B2Http._translate_errors(lambda: response)
-        self.assertIs(response, actual)
+        assert response == actual
 
     def test_partial_content(self):
         response = MagicMock()
         response.status_code = 206
         actual = B2Http._translate_errors(lambda: response)
-        self.assertIs(response, actual)
+        assert response == actual
 
     def test_b2_error(self):
         response = MagicMock()
         response.status_code = 503
         response.content = b'{"status": 503, "code": "server_busy", "message": "busy"}'
-        with self.assertRaises(ServiceError):
+        with pytest.raises(ServiceError):
             B2Http._translate_errors(lambda: response)
 
     def test_broken_pipe(self):
@@ -66,7 +66,7 @@ class TestTranslateErrors(TestBase):
                 )
             )
 
-        with self.assertRaises(BrokenPipe):
+        with pytest.raises(BrokenPipe):
             B2Http._translate_errors(fcn)
 
     def test_unknown_host(self):
@@ -77,14 +77,14 @@ class TestTranslateErrors(TestBase):
                 )
             )
 
-        with self.assertRaises(UnknownHost):
+        with pytest.raises(UnknownHost):
             B2Http._translate_errors(fcn)
 
     def test_connection_error(self):
         def fcn():
             raise requests.ConnectionError('a message')
 
-        with self.assertRaises(B2ConnectionError):
+        with pytest.raises(B2ConnectionError):
             B2Http._translate_errors(fcn)
 
     def test_connection_reset(self):
@@ -94,14 +94,14 @@ class TestTranslateErrors(TestBase):
         def fcn():
             raise SysCallError('(104, ECONNRESET)')
 
-        with self.assertRaises(ConnectionReset):
+        with pytest.raises(ConnectionReset):
             B2Http._translate_errors(fcn)
 
     def test_unknown_error(self):
         def fcn():
             raise Exception('a message')
 
-        with self.assertRaises(UnknownError):
+        with pytest.raises(UnknownError):
             B2Http._translate_errors(fcn)
 
     def test_too_many_requests(self):
@@ -109,7 +109,7 @@ class TestTranslateErrors(TestBase):
         response.status_code = 429
         response.headers = {'retry-after': 1}
         response.content = b'{"status": 429, "code": "Too Many requests", "message": "retry after some time"}'
-        with self.assertRaises(TooManyRequests):
+        with pytest.raises(TooManyRequests):
             B2Http._translate_errors(lambda: response)
 
     def test_invalid_json(self):
@@ -129,7 +129,7 @@ class TestTranslateErrors(TestBase):
         response.content = b'<?xml version="1.0" encoding="UTF-8" standalone="yes"?>'
         response.url = 'https://s3.us-west-000.backblazeb2.com'
 
-        with self.assertRaises(PotentialS3EndpointPassedAsRealm):
+        with pytest.raises(PotentialS3EndpointPassedAsRealm):
             B2Http._translate_errors(lambda: response)
 
 

--- a/test/unit/b2http/test_b2http.py
+++ b/test/unit/b2http/test_b2http.py
@@ -132,6 +132,16 @@ class TestTranslateErrors:
         with pytest.raises(PotentialS3EndpointPassedAsRealm):
             B2Http._translate_errors(lambda: response)
 
+    @pytest.mark.apiver(to_ver=2)
+    def test_bucket_id_not_found(self):
+        from b2sdk.v2.exception import BucketIdNotFound, v3BucketIdNotFound
+
+        def fcn():
+            raise v3BucketIdNotFound('bucket_id')
+
+        with pytest.raises(BucketIdNotFound):
+            B2Http._translate_errors(fcn)
+
 
 def test_b2_error__nginx_html():
     """

